test/test_torch.py

```diff
@@ -313,6 +313,38 @@ def test_dim_reduction_less_than_64(self):
         with self.assertRaisesRegex(RuntimeError, "PyTorch doesn't support reduction operations for dim>=64"):
             torch.sum(x, -1)
 
+    def _test_logaddexp(self, base2=False):
+        import numpy as np
+
+        if base2:
+            gt_func = np.logaddexp2
+            our_func = torch.logaddexp2
+        else:
+            gt_func = np.logaddexp
+            our_func = torch.logaddexp
+
+        a = torch.randn(51, 2)
+        b = torch.randn(51, 2)
+        gt = gt_func(a.numpy(), b.numpy())
+        ours = our_func(a, b)
+        self.assertEqual(ours.shape, gt.shape)
+        self.assertTrue(np.allclose(ours.numpy(), gt))
+
+        # numerical stability
+        a *= 10000
+        b *= 10000
+        gt = gt_func(a.numpy(), b.numpy())
+        ours = our_func(a, b)
+        self.assertTrue(np.allclose(ours.numpy(), gt))
+
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
```

Comment:
You should move these tests into TestTorchDeviceType so you can run them on both the CPU and GPU. Right now you're just testing the CPU. Then see the helper function "_np_compare," which you can use to simplify your tests, and the @dtypes decorator so you can test multiple dtypes. Since you're only enabling logaddexp on float types, you should test at least torch.long (assert throws RuntimeError), torch.float32, and torch.complex64 (assert throws RuntimeError). That way if someone implements logaddexp for complex64, for example, they'll know to update this test. 

_np_compare works by taking values to test at. Your value generation is OK but you should add some more "interesting" values like -math.pi, 0, and math.pi, as well as extremal values like nan, -inf, and inf.